yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/reader/Tokenizer.scala

```diff
@@ -306,5 +306,5 @@ private[yaml] class Scanner(str: String) extends Tokenizer {
   def skipUntilNextChar() =
     while (in.isWhitespace) do in.skipCharacter()
 
-  private def skipComment(): Unit = while !in.isNewline do in.skipCharacter()
+  private def skipComment(): Unit = while (in.peek().isDefined && !in.isNewline) do in.skipCharacter()
```

Comment:
I think, that you should modify `in.isNewline` method and them check, if peek() exists.

Reply:
It's not enough. We are skipping until we find a newline character, but for some input that will never happen. Being a new line and being defined is a different thing.